tensorflow/core/graph/mkl_layout_pass_test.cc

```diff
@@ -185,308 +226,375 @@ TEST_F(MklLayoutPassTest, Basic) {
-      "node { name: 'D' op: 'Input'}"
-      "node { name: 'E' op: 'BiasAdd'"
-      " attr { key: 'T'                value { type: DT_FLOAT } }"
-      " attr { key: 'data_format'      value { s: 'NCHW' } }"
-      " input: ['C', 'D'] }"
-      "node { name: 'Y' op: 'Input'}"
-      "node { name: 'Z' op: 'Zeta'"
-      " attr {key: 'T'                 value { type: DT_FLOAT } }"
-      " input: ['E', 'Y']}");
-  EXPECT_EQ(DoMklLayoutOptimizationPass(),
-            "A(Input);B(Input);D(Input);DMT/_0(Const);DMT/_1(Const);"
-            "DMT/_2(Const);E(_MklConv2DWithBias);Y(Input);Z(Zeta)|A->E;"
-            "A:control->DMT/_0:control;A:control->DMT/_1:control;"
-            "A:control->DMT/_2:control;B->E:1;D->E:2;DMT/_0->E:3;DMT/_1->E:4;"
-            "DMT/_2->E:5;E->Z;Y->Z:1");
-}
+#define REGISTER_TEST(NAME, T, INPUT)                                      \
+  TEST_F(MklLayoutPassTest, NAME##_##T) {                                  \
+    CHECK_EQ(kTensorOrdering, MklTfTensorOrdering::TENSORS_CONTIGUOUS);    \
+    InitGraph("node { name: 'A' op: '" #INPUT "'}"                         \
+              "node { name: 'B' op: '" #INPUT "'}"                         \
+              "node { name: 'C' op: 'Conv2D'"                              \
+              " attr { key: 'T'                value { type:" #T " } }"    \
+              " attr { key: 'data_format'      value { s: 'NCHW' } }"      \
+              " attr { key: 'use_cudnn_on_gpu' value { b: false } }"       \
+              " attr { key: 'strides'          value { list: {i: 1, i:1, " \
+              "i:1, i:1} } }"                                              \
+              " attr { key: 'padding'          value { s: 'SAME' } }"      \
+              " attr { key: 'dilations'        value { list: {i: 1, i:1, " \
+              "i:1, i:1} } }"                                              \
+              " input: ['A', 'B']}"                                        \
+              "node { name: 'D' op: '" #INPUT "'}"                         \
+              "node { name: 'E' op: 'BiasAdd'"                             \
+              " attr { key: 'T'                value { type:" #T " } }"    \
+              " attr { key: 'data_format'      value { s: 'NCHW' } }"      \
+              " input: ['C', 'D'] }"                                       \
+              "node { name: 'Y' op: '" #INPUT "'}"                         \
+              "node { name: 'Z' op: 'Zeta'"                                \
+              " attr {key: 'T'                 value { type:" #T " } }"    \
+              " input: ['E', 'Y']}");                                      \
+                                                                           \
+    EXPECT_EQ(                                                             \
+        DoMklLayoutOptimizationPass(),                                     \
+        "A(" #INPUT ");B(" #INPUT ");D(" #INPUT                            \
```

Comment:
Nit: Would you mind filling each line as much as possible (without ugly word breaks) too? (Totally okay if you don't want to.)

Reply:
Yeah, fixed this. I agree that lines were looking ugly otherwise.